pkg/ocm/billing.go

```diff
@@ -34,22 +36,61 @@ func (c *Client) GetBillingAccounts() ([]string, error) {
 		return nil, handleErr(quotaCostResponse.Error(), err)
 	}
 
-	var billingAccounts []string
+	var billingAccounts []*v1.CloudAccount
 	for _, item := range quotaCostResponse.Items().Slice() {
-		for _, account := range item.CloudAccounts() {
-			if account.CloudProviderID() == "aws" && !helper.Contains(billingAccounts, account.CloudAccountID()) {
-				billingAccounts = append(billingAccounts, account.CloudAccountID())
-			}
-		}
+		billingAccounts = append(billingAccounts, item.CloudAccounts()...)
 	}
 
 	if len(billingAccounts) == 0 {
-		return billingAccounts, errors.New("no billing accounts found")
+		return billingAccounts, errors.New("No billing account associated. " +
+			"Go to https://docs.openshift.com/rosa/rosa_architecture/rosa-understanding.html" +
+			" to learn how to associate your billing account." +
+			" You must associate a billing account to continue.")
 	}
 
 	return billingAccounts, nil
 }
 
+func GenerateBillingAccountsList(cloudAccounts []*v1.CloudAccount) []string {
+	var billingAccounts []string
+	for _, account := range cloudAccounts {
+		if account.CloudProviderID() == "aws" && !helper.ContainsPrefix(billingAccounts, account.CloudAccountID()) {
+			contracts, hasContract := account.GetContracts()
+			var contractString string
+			if hasContract {
+				dimensions := contracts[0].Dimensions()
+				for _, dimension := range dimensions {
+					if HasValidContracts(dimension) {
+						contractString = " [Contract enabled]"
+						break
+					}
+				}
+			}
+			billingAccounts = append(billingAccounts, account.CloudAccountID()+contractString)
+		}
+	}
+	return billingAccounts
+
+}
+
+func GetDimensionValue(dimension *v1.ContractDimension, resource string) int {
+	var value = 0
+	if dimension.Name() == resource {
+		value, _ = strconv.Atoi(dimension.Value())
```

Comment:
I'm not sure we can assume that dimensions are always going to be convertible  to integers (except for four_vcpu_hour and control_plane) - probably better to deal with it separately.

Reply:
So far it's only used for four_vcpu_hour and control_plane, but I got you, we can make it more generic. Just to confirm, are you suggesting that we add two cases for those two types. For the rest, we don't do the number conversion, but just return the value?

Reply:
@chenz4027 how about GetNumsOfVPCsAndClusters(takes in a dimension slice) return (int, int) first being numberOfVPCs and second being numberOfClusters.
In the function iterate over the dimensions slice and set the numbers according and return them at the end. return 0s as fall back.
I think this might help address @cristianoveiga other concern about HasValidContracts as well.

Reply:
Thanks @mikeshng @cristianoveiga , updated the function as https://github.com/openshift/rosa/pull/1500/files#diff-54e259e8c0b289af4e61624a5e697f0285de9d9e0e14375c7d4fa82c772038eaR69